src/radix_tree.rs

```diff
@@ -0,0 +1,164 @@
+use crate::vwmap::NamespaceDescriptor;
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub(crate) struct NamespaceDescriptorWithHash {
+    pub(crate) descriptor: NamespaceDescriptor,
+    pub(crate) hash_seed: u32,
+}
+
+impl NamespaceDescriptorWithHash {
+    pub(crate) fn new(descriptor: NamespaceDescriptor, hash_seed: u32) -> Self {
+        Self {
+            descriptor,
+            hash_seed
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+struct RadixTreeNode {
+    children: Vec<Option<RadixTreeNode>>,
+    value: Option<NamespaceDescriptorWithHash>,
+}
+
+impl Default for RadixTreeNode {
+    fn default() -> Self {
+        Self {
+            children: vec![None; 256],
```

Comment:
Why 256?

Reply:
256 UTF-8 characters ;)

Reply:
Surely UTF-8 doesn't enable just 2^8 chars though? Chars can be encoded not by a single byte (multi-byte chars) - does that play a role here? (>2e6 possible chars afaik) Further, are there not only 128 characters represented by one byte in UTF-8 (most likely this has no impact on the implementation of course)

Reply:
Only took into account the ASCII ones as we use those primarily.